File: sdk/examples/ready.mjs
#!/usr/bin/env node
/**
 * Zero-friction Node gate — run from repo after `cd sdk && npm ci && npm run build`:
 *   node examples/ready.mjs
 *   node examples/ready.mjs --origin https://zakai-3uxj.vercel.app
 *
 * Prefer: npm run ready  (same gate, packaged bin)
 */
import { spawnSync } from "node:child_process";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

const here = dirname(fileURLToPath(import.meta.url));
const bin = join(here, "..", "dist", "ready-bin.js");
const originIdx = process.argv.indexOf("--origin");
const args = originIdx >= 0 ? process.argv.slice(originIdx) : [];
const r = spawnSync(process.execPath, [bin, ...args], { stdio: "inherit" });
process.exit(r.status ?? 1);

File: sdk/python/examples/ready.py
#!/usr/bin/env python3
"""
Zero-friction Python gate — from sdk/python:

  pip install -e '.[crypto]'
  python examples/ready.py
  python examples/ready.py --origin https://zakai-3uxj.vercel.app

Prefer: zakai-mandate-ready  (same entrypoint)
"""

from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1] / "src"
sys.path.insert(0, str(ROOT))

from zakai_mandate.verify import main  # noqa: E402

if __name__ == "__main__":
    # Default to --ready when no flags (same as zakai-mandate-ready).
    if len(sys.argv) == 1:
        sys.argv.append("--ready")
    raise SystemExit(main())
